src/mainwindow.cpp

```diff
@@ -2516,7 +2516,10 @@ void MainWindow::on_actionTileset_Editor_triggered()
         this->tilesetEditor = new TilesetEditor(this->editor->project, this->editor->map, this);
         connect(this->tilesetEditor, SIGNAL(tilesetsSaved(QString, QString)), this, SLOT(onTilesetsSaved(QString, QString)));
         connect(this->tilesetEditor, &QObject::destroyed, [=](QObject *) { this->tilesetEditor = nullptr; });
-        this->tilesetEditor->setAttribute(Qt::WA_DeleteOnClose);
```

Comment:
By removing `Qt::WA_DeleteOnClose`, is there now a memory leak?  I thought I remember setting that on purpose.

Reply:
Perhaps calling `closeSupplementaryWindows()` somewhere along the line when porymap is being closed (probably the `MainWindow` destructor) is sufficient.
Assuming there are no issues in using the same `TilesetEditor` object for the entire session.

Reply:
I remember checking that it calls the destructor because I had the same thought, and it does as far as I remember. But this was also 9 days ago, so I'll re-test this and let you know.

Reply:
I guess since it is created as a child of `MainWindow`, it would already get cleaned up by the destructor.

Reply:
Okay, I just verified, and all the destructors get called whether those windows are still open or not. But I also noticed a slight bug that allows you to close the `TilesetEditor` while the `PaletteEditor` remains open, effectively allowing you to edit the tileset while the window is 'closed'. Then if you re-open the `TilesetEditor` and close it again, it then will prompt you to save those changes. I'll fix that.